data_steward/utils/bq.py

```diff
@@ -193,6 +193,50 @@ def define_dataset(project_id, dataset_id, description, label_or_tag):
     return dataset
 
 
+def update_labels_and_tags(project_id,
+                           dataset_id,
+                           label_or_tag,
+                           set_or_update=True):
+    """
+    Updates labels or tags in dataset if not set or needing to be updated
+    or overwrites existing labels or tags in the dataset
+
+    :param project_id:  string name to identify the project
+    :param dataset_id:  string name to identify the dataset
+    :param label_or_tag:  labels for the dataset = Dict[str, str]
+                          tags for the dataset = Dict[str, '']
+    :param set_or_update:  flag to signal if label_or_tag is to be either
+                             set or updated (True as default) or overwritten (False)
+
+    :raises:  RuntimeError if parameters are not specified
+
+    :return:  a dataset reference object
+    """
+    if not project_id:
+        raise RuntimeError(
+            "Specify the project_id for the project containing the dataset")
+
+    if not dataset_id:
+        raise RuntimeError("Provide a dataset_id")
+
+    if not label_or_tag:
+        raise RuntimeError("Please provide a label or tag")
+
+    client = get_client(project_id)
+    dataset_id = f"{project_id}.{dataset_id}"
+    dataset = client.get_dataset(dataset_id)
+
+    if set_or_update is True:
+        # will set or update labels and/or tags
+        dataset.labels = label_or_tag
+        dataset = client.update_dataset(dataset, ['labels'])
```

Comment:
So in the future it might be best to create an `update_dataset()` function in `bq` instead of relying on the `update_dataset()` from the bq client library?

In order to not use `client.update_dataset()` would be to just append the new labels/tag to the already existing list of labels/tags right?

Reply:
I'd like @lrwb-aou 's opinion on it, but yes, I'm considering if we will want a `bq.update_dataset()` which wraps `client.update_dataset()`. 

So as-is the use case I have in mind might look like this:
```python
dataset_items = client.list_datasets()
for dataset_item in dataset_items:
  dataset = client.get_dataset(dataset_item.dataset_id)
  standard_labels = get_standard_labels(dataset.dataset_id)

  #  raise an error if labels exist
  dataset = bq.update_labels_and_tags(client, project_id, dataset.dataset_id, standard_labels)

  #  overwrite any existing labels
  dataset = bq.update_labels_and_tags(client, project_id, dataset.dataset_id, standard_labels, overwrite_ok=True)

  #  preserve any existing labels
  result_labels = {**dataset.labels, **standard_labels}
  dataset = bq.update_labels_and_tags(client, project_id, dataset.dataset_id, result_labels, overwrite_ok=True)

  #  no labels raise an error
  dataset = bq.update_labels_and_tags(client, project_id, dataset.dataset_id, labels={})
```

Or it might look like this if we end up wrapping `update_dataset()` 
```python
dataset_items = client.list_datasets()
for dataset_item in dataset_items:
  dataset = client.get_dataset(dataset_item.dataset_id)
  standard_labels = get_standard_labels(dataset.dataset_id)

  #  overwrite any existing labels
  dataset.labels = standard_labels

  #  preserve any existing labels
  dataset.labels = {**dataset.labels, **standard_labels}

  #  no labels raise an error on bq.update_dataset()
  dataset.labels = {}
  dataset = bq.update_dataset(client, 
                              fields=['labels']  #  ew, I know
                              dataset=dataset)
```

Reply:
@mark-velez and @dcampbell-vumc I would argue that ticket 758 is flawed.  The ticket was written when we were still thinking of using a wrapper for the client library like we did with bq_utils.  However, if curation adheres to requiring modules to retain a client object and use that client, the scope of this ticket changes to ensuring the business logic requirements of tagging a dataset are met.  So, bq might contain a `get_standard_labels` function that returns a dictionary.  It might also contain `update_labels(dataset_id, existing_labels, new_labels)` to make sure that new labels don't overwrite existing labels, labels follow business practices, labels include dataset names, etc..  The output would be a dictionary and the client would call `update_dataset`.  So something similar to:
```
dataset = client.get_dataset(dataset_id)
existing_labels = dataset.labels
new_labels = {'test': '', 'random_key': 'random_value'}
updated_labels = bq.update_labels(dataset_id, existing_labels, new_labels)
client.update_dataset(dataset, fields=['labels'])
```

Reply:
@lrwb-aou and @mark-velez: I don't mind this ticket being rescoped and or broken into smaller tickets focusing on creating a `get_standard_labels` function and to modify this `update_labels` function. I am unsure on who exactly makes that call but judging from previous discussions about bq, I think it's best if we go the passing the client object route that was agreed upon.

Reply:
@dcampbell-vumc I think `get_standard_labels()` is just a hypothetical function to illustrate future usage of `bq.update_labels()` and not in scope for this ticket. I understand @lrwb-aou to be saying that `bq.update_labels()` is a function without side-effects- it does not actually make an API call and update the dataset and so shouldn't need a client object. It is just a utility that helps us put together the label/tag dictionary in the appropriate way. Let me know if this does [not] make sense. Happy to discuss over phone too.